libs/clients/middlewares/src/lib/createEnhancedFetch.ts

```diff
@@ -129,7 +129,10 @@ export const createEnhancedFetch = (
 
   builder.wrap(withAgent, {
     clientCertificate,
-    agentOptions,
+    agentOptions: {
+      timeout: typeof timeout === 'number' ? timeout : undefined,
```

Comment:
If `timeout` is false, there should be no timeout. Can we disable agentkeepalive's timeout?

Also, now we have two implementations for the `timeout` argument. Should maybe get rid of `withTimeout`?

 Or should we use